src/tuple.cr

```diff
@@ -347,6 +347,38 @@ struct Tuple
     {% end %}
   end
 
+  # Yields each of the elements in this tuple in reverse order.
+  #
+  # ```
+  # tuple = {1, "hello", 'x'}
+  # tuple.reverse_each do |value|
+  #   puts value
+  # end
+  # ```
+  #
+  # Output:
+  #
+  # ```text
+  # 'x'
+  # "hello"
+  # 1
+  # ```
+  def reverse_each
+    (size - 1).downto(0) do |i|
+      yield self[i]
+    end
```

Comment:
Have a look at `Tuple#each` etc above, using the macro language should give the compiler easier to optimize code :)